streamer/src/nonblocking/quic.rs

```diff
@@ -690,33 +690,35 @@ async fn setup_connection(
         match connecting_result {
             Ok(new_connection) => {
                 debug!("Got a connection {from:?}");
-                if !rate_limiter.is_allowed(&from.ip()) {
-                    debug!("Reject connection from {from:?} -- rate limiting exceeded");
+                // now that we have observed the handshake we can be certain
+                // that the initiator owns an IP address, we can update rate
+                // limiters on the server
+                if overall_connection_rate_limiter.consume_tokens(1).is_err() {
```

Comment:
We should not do overall_connection rate limit first. A noisy connection maker will crowd out others and make them harder to make connections. I think we still should do per-ip address first.

Reply:
Overall limiter is cheaper to check. Does order of checks make any difference if we check both anyway?

Reply:
Yes. Imagine a noisy connector quickly passing through the overall gate but bring the limit close to the limit. The innocent connector will have trouble to go through this overall gate. Cause kind of DOS. We used to have the overall limiter first and keyed limiter after it. We explicitly changed the position due to that concern.

Reply:
Yes but if overall limit is exhausted we will drop connection no matter the order of checking.

Reply:
Of course you are right! Apologies for being dense here.

Reply:
Fixed in https://github.com/anza-xyz/agave/pull/8154/commits/a780c9a95738b92a638eb409858a0c8b8d40fce2